modules/proxy-server/main.tf

```diff
@@ -0,0 +1,61 @@
+########################################################################################################################################################################################################################
+#
+# Creates an nginx proxy server with associated security groups to proxy a web server
+#
+########################################################################################################################################################################################################################
+
+# ---------------------------------------------------------------------------------------------------------------------
+# THESE TEMPLATES REQUIRE TERRAFORM VERSION 0.8 AND ABOVE
+# ---------------------------------------------------------------------------------------------------------------------
+
+terraform {
+  required_version = ">= 0.9.3"
+}
+
+resource "aws_instance" "proxy_instance" {
+  instance_type         = "${var.instance_type}"
+  ami                   = "${var.ami_id}"
+  key_name              = "${var.ssh_key_name}"
+  subnet_id             = "${var.subnet_id}"
+  associate_public_ip_address = "${var.associate_public_ip_address}"
+  vpc_security_group_ids = ["${aws_security_group.proxy_security_group.id}", "${var.aws_security_group_ids}"]
+  user_data              = "${var.user_data == "" ? data.template_file.proxy_user_data.rendered : var.user_data}"
+  tags {
```

Comment:
Instead of taking a SAN tag as a specific variable, lets use a tag variable:
```tf
variable "tags" {
  description = "Tags to apply to the EC2 instance. Name will be applied by default."
  type        = "map"
  default     = {}
}
```

In the instance when referring to the variable value use
```tf
...
tags = "${merge(var.tags, map("Name", "${var.instance_name}"))}"
...
```
This will let us add any number of tags to the instance